trackintel/model/util.py

```diff
@@ -1,4 +1,102 @@
 from functools import partial, update_wrapper
+import trackintel as ti
+import numpy as np
+import pandas as pd
+from trackintel.geogr.distances import calculate_haversine_length
+
+
+def speed_positionfixes(positionfixes):
+    """
+    Compute speed per positionfix
+
+    Parameters
+    ----------
+    positionfixes : GeoDataFrame (as trackintel positionfixes)
+        The positionfixes have to follow the standard definition for positionfixes DataFrames.
+
+    Returns
+    -------
+    pfs: GeoDataFrame (as trackintel positionfixes)
+        The original positionfixes with a new column ``[`speed`]``. The speed is given in km/h
+
+    Notes
+    -----
+    The speed at one positionfix is computed from the distance and time since the previous positionfix. For the first
+    positionfix, the speed is set to the same value as for the second one.
+    """
+    pfs = positionfixes.copy()
+    if_planar_crs = ti.geogr.distances.check_gdf_planar(pfs)
+
+    # get next location and time
+    pfs["prev_geom"] = pfs["geom"].shift(1)
+    pfs.loc[pfs.index[0], "prev_geom"] = pfs.loc[pfs.index[0], "geom"]
+    pfs["prev_tracked_at"] = pfs["tracked_at"].shift(1)
+
+    # get distance and time difference
+    if if_planar_crs:
+        dist_function = lambda point: point.geom.distance(point.prev_geom)
+    else:
+        dist_function = lambda point: ti.geogr.point_distances.haversine_dist(
+            point.geom.x, point.geom.y, point.prev_geom.x, point.prev_geom.y
+        )[0]
+    pfs["dist"] = pfs.apply(dist_function, axis=1)
+    pfs["time_diff"] = pfs.apply(lambda x: x.tracked_at - x.prev_tracked_at, axis=1).dt.total_seconds()
+
+    # compute speed and convert to kmh
+    pfs["speed"] = 3.6 * pfs["dist"] / pfs["time_diff"]
+    # The first point speed is imputed
+    pfs.loc[pfs.index[0], "speed"] = pfs.iloc[1]["speed"]
+
+    pfs.drop(columns=["prev_geom", "prev_tracked_at", "dist", "time_diff"], inplace=True)
+
+    return pfs
+
+
+def speed_triplegs(triplegs):
```

Comment:
Docstring missing. You can copy the docstring from below and slightly change the description